src/tests/ftest/io/EightServers_mpiio.py

```diff
@@ -0,0 +1,168 @@
+from daos_api import DaosContext, DaosPool, DaosApiError
+
+class EightServers(Test):
+    """
+    Test class Description: Runs IOR with 8 servers.
+
+    """
+
+    def __init__(self, *args, **kwargs):
+
+        super(EightServers, self).__init__(*args, **kwargs)
+
+        self.basepath = None
+        self.server_group = None
+        self.Context = None
+        self.pool = None
+        self.np = None
+        self.hostlist_servers = None
+        self.hostfile_servers = None
+        self.hostlist_clients = None
+        self.hostfile_clients = None
+
+    def setUp(self):
+        # get paths from the build_vars generated by build
+        with open('../../../.build_vars.json') as f:
+            build_paths = json.load(f)
+        self.basepath = os.path.normpath(build_paths['PREFIX']  + "/../")
+        print "<<{}>>".format(self.basepath)
+        self.server_group = self.params.get("server_group", '/server/', 'daos_server')
+
+        # setup the DAOS python API
+        self.Context = DaosContext(build_paths['PREFIX'] + '/lib/')
+
+        self.hostlist_servers = self.params.get("test_servers", '/run/hosts/test_machines/*')
+        self.hostfile_servers = WriteHostFile.WriteHostFile(self.hostlist_servers, self.workdir)
+        print("Host file servers is: {}".format(self.hostfile_servers))
+
+        self.hostlist_clients = self.params.get("test_clients", '/run/hosts/test_machines/*')
+        self.np = self.params.get("np", '/run/ior/client_processes/*')
+        self.hostfile_clients = WriteHostFile.WriteHostFile(self.hostlist_clients, self.workdir)
+        print("Host file clients is: {}".format(self.hostfile_clients))
+
+        ServerUtils.runServer(self.hostfile_servers, self.server_group, self.basepath)
+
+    def tearDown(self):
+        try:
+            if self.pool is not None and self.pool.attached:
+                self.pool.destroy(1)
+        finally:
+            ServerUtils.stopServer(hosts=self.hostlist_servers)
+
+    def executable(self, iorflags=None):
+        """
+        Executable function to run ior for ssf and fpp
+        """
+
+        # parameters used in pool create
+        createmode = self.params.get("mode", '/run/pool/createmode/*/')
+        createuid = os.geteuid()
+        creategid = os.getegid()
+        createsetid = self.params.get("setname", '/run/pool/createset/')
+        createscm_size = self.params.get("scm_size", '/run/pool/createsize/')
+        createnvme_size = self.params.get("nvme_size", '/run/pool/createsize/')
+        createsvc = self.params.get("svcn", '/run/pool/createsvc/')
+        iteration = self.params.get("iter", '/run/ior/iteration/')
+        block_size = self.params.get("b", '/run/ior/transfersize_blocksize/*/')
+        transfer_size = self.params.get("t", '/run/ior/transfersize_blocksize/*/')
+        expected_result = 'PASS'
+
+        try:
+            # Do not need slots in hostfile
+            with open(self.hostfile_clients) as f:
```

Comment:
Change the WriteHostFile function to allow for this with slots=None or something.

Reply:
Done